File: app/domain/PreservationType.scala
package domain

object PreservationType extends Enumeration {
  type PreservationType = Value
  val FrozenSpecimen = Value("Frozen Specimen")
  val RnaLater = Value("RNA Later")
  val FreshSpecimen = Value("Fresh Specimen")
  val Slide = Value("Slide")
}

object PreservationTemperatureType extends Enumeration {
  type PreservationTemperatureType = Value
  val Plus4celcius = Value("4 C")
  val Minus20celcius = Value("-20 C")
  val Minus80celcius = Value("-80 C")
  val Minus180celcius = Value("-180 C")
  val RoomTemperature = Value("Room Temperature")
}

File: app/assets/javascripts/domain/PreservationType/PreservationType.js
/**
 * AngularJS Constants used for defining specimen types.
 *
 * @namespace domain.PreservationType
 *
 * @author Nelson Loyola <loyola@ualberta.ca>
 * @copyright 2018 Canadian BioSample Repository (CBSR)
 */

/**
 * A standardised set of methods for preserving and storing {@link domain.participants.Specimen Specimens}.
 *
 * Potential examples include: frozen specimen, RNA later, fresh specimen, slide, etc.
 *
 * @enum {string}
 * @memberOf domain.PreservationType
 */
const PreservationType = {
  FROZEN_SPECIMEN: 'Frozen Specimen',
  RNA_LATER:       'RNA Later',
  FRESH_SPECIMEN:  'Fresh Specimen',
  SLIDE:           'Slide'
};


export default ngModule => ngModule.constant('PreservationType', PreservationType)
